File: server/models/notes.js
const mongoose = require("mongoose");
const taskSchema = require("./task");


const notesSchema = mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    title: {
        type: String,
        required: true,
        trim: true,
        default: "Untitled Note"
    },
    description: {
        type: String,
        trim: true,
        default: ""
    },
    tasks: [taskSchema.taskSchema],

},
    { timestamps: true }
);

const Notes = mongoose.model("Notes", notesSchema);
module.exports = Notes;

File: lib/models/notes.dart
import 'package:node_todo_app/models/tasks.dart';
import 'package:provider/provider.dart';

class Notes {
  String? id;
  String userId;
  String title;
  String description;
  List<Task> tasks;
  String? createdAt;
  String? updatedAt;

  Notes({
    required this.userId,
    this.id,
    this.title = "Untitled Note",
    this.description = "",
    this.tasks = const [],
    this.createdAt,
    this.updatedAt,
  });

  factory Notes.fromJson(Map json) {
    return Notes(
      id: json["_id"],
      userId: json['userId'],
      title: json['title'] ?? "Untitled Note",
      description: json['description'] ?? "",
      tasks: (json['tasks'] as List<dynamic>?)
              ?.map((taskJson) => Task.fromJson(taskJson))
              .toList() ??
          [],
      createdAt: json["createdAt"],
      updatedAt: json["updatedAt"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'title': title,
      'description': description,
      'tasks': tasks.map((task) => task.toJson()).toList(),
    };
  }
}
